snap/patches/nova/0001-remotefs-http-driver.patch

```diff
@@ -0,0 +1,246 @@
+ 
+ libvirt_volume_vzstorage_opts = [
+diff --git a/nova/virt/libvirt/volume/remotefs.py b/nova/virt/libvirt/volume/remotefs.py
+index 8ddcaa1fcc..8096c77903 100644
+--- a/nova/virt/libvirt/volume/remotefs.py
++++ b/nova/virt/libvirt/volume/remotefs.py
+@@ -24,6 +24,8 @@ from oslo_utils import importutils
+ import nova.conf
+ import nova.privsep.fs
+ from nova import utils
++from nova.virt.libvirt.volume.remotefs_http import HttpDriver as _HttpDriverImpl
++HttpDriver = _HttpDriverImpl
+ 
+ LOG = logging.getLogger(__name__)
+ 
+diff --git a/nova/virt/libvirt/volume/remotefs_http.py b/nova/virt/libvirt/volume/remotefs_http.py
+new file mode 100644
+index 0000000000..ed983e2963
+--- /dev/null
++++ b/nova/virt/libvirt/volume/remotefs_http.py
+@@ -0,0 +1,174 @@
++import os
++import ssl
++import tempfile
++from typing import Tuple
++
++from oslo_log import log as logging
++
++import nova.conf
++
++
++LOG = logging.getLogger(__name__)
++CONF = nova.conf.CONF
++
++LOG.debug("[webdav] remotefs_http module loaded")
++
++
++def _debug_log_path(label: str, path: str) -> str:
++    LOG.debug("[webdav] %s path=%s", label, path)
++    return path
++
++
++def _parse_remote_dst(dst: str) -> Tuple[str, str]:
++    # Expected format: "<host>:<absolute_path>"
++    # NOTE: Only IPv4 addresses are supported; IPv6 bracket notation is not handled.
++    if ":" not in dst:
++        raise ValueError(f"Invalid remote path format, expected 'host:/abs/path', got: {dst}")
++    host, path = dst.split(":", 1)
++    if not path.startswith("/"):
++        raise ValueError(f"Destination path must be absolute, got: {path}")
++    return host, path
++
++
++def _ensure_trailing_slash(path: str) -> str:
++    if not path or path.endswith("/"):
++        return path
++    return f"{path}/"
++
++
++def _base_url(host: str) -> str:
++    port = getattr(CONF.libvirt, "file_server_port", 10099)
++    return f"https://{host}:{port}"
++
++
++def _new_client(host: str):
++    # Lazy import to keep dependency surface localized
++    from webdav4.client import Client
++
++    base_url = _base_url(host)
++
++    # TLS verification and client certs for mTLS
++    ca_file = getattr(CONF.libvirt, "file_server_ca_file", None) or True
++    cert_file = getattr(CONF.libvirt, "file_server_client_cert_file", "") or None
++    key_file = getattr(CONF.libvirt, "file_server_client_key_file", "") or None
++    client_cert = (cert_file, key_file) if cert_file and key_file else None
++
++    kwargs = {}
++    # Handle SSL verification: verify certificate chain but not hostname
++    # (since we may connect via IP address, not hostname)
++    if ca_file and ca_file is not True:
++        ssl_context = ssl.create_default_context(cafile=ca_file)
++        ssl_context.check_hostname = False
++        kwargs["verify"] = ssl_context
++    elif ca_file is True:
++        ssl_context = ssl.create_default_context()
++        ssl_context.check_hostname = False
++        kwargs["verify"] = ssl_context
++    else:
```

Comment:
Confused when this will get executed since L#114 returns True if file_server_ca_file is not set